Tests/SentryTests/Transaction/SentryTracerTests.swift

```diff
@@ -52,8 +52,6 @@ class SentryTracerTests: XCTestCase {
 #endif // os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
         
         init() {
-            dispatchQueue.blockBeforeMainBlock = { false }
```

Comment:
I don't know why this was configured this way. It essentially led to test cases never actually initializing timeout timers, so calling TestSentryNSTimerFactory.fire() was a no-op.

Reply:
I think that was a hacky workaround for skipping initializing the timer, we can remove it safely. I also remove the other invocations with: https://github.com/getsentry/sentry-cocoa/pull/4826/commits/3ab75ee75a65c6810ccb94e2f3a82a2c72df4c6b